x-pack/solutions/observability/plugins/streams/server/lib/streams/helpers/validate_stream.ts

```diff
@@ -97,14 +96,14 @@ export function validateStreamChildrenChanges(
   }
 }
 
-export function validateStreamLifecycle(definition: WiredStreamDefinition, isServerless: boolean) {
+export function validateStreamLifecycle(definition: StreamDefinition, isServerless: boolean) {
```

Comment:
should probably be `IngestStreamLifecycle` (this will get relevant with Chris' PR)

Reply:
We also check the type of the definition in this function, I can move it out as it's a temporary check until ilm for unwired is implemented, but could you clarify how passing a lifecycle will help ?

Reply:
Didn't check where it's used, you are right, going by `StreamDefinition` makes the most sense.